### PR TITLE
chore: Add basic cursor rules

### PR DESCRIPTION
**Pull request by sfc-gh-jcieslak**
## Changes
- Introduce base cursor rules and rules for sdk development (mainly for the use case of adding/maintaining sdk objects used later in resource implementations)
- Basic rules are divided into two. One for defining what rules should look like a how they should be generally structured. The second one is for automatic cursor analysis and rule improvement (this needs to be monitored with increased usage as I didn't notice it working, but it may be because I was mainly working on rules not the actual implementations).
- The sdk rules were mainly added with new SDK object generation in mind. I have tested it a few times with different objects and I can confirm there was some improvement once the rules were added (I also improved them a bit when generating different objects).

## Next prs
- I wanted to complete the whole use case of adding support for the new object by adding the resource (and maybe data source) part, which includes: resource schema generation and implementing CRUD functions, generating models and asserts, generating and writing schema for SHOW/DESC outputs, writing common acceptance test scenarios, etc.

### CODE REVIEW

File: .cursor/rules/sdk-development.mdc [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+- Enum conversion methods and error cases
+- Custom helper functions and utilities
+- SQL generation edge cases
+- Complex validation logic not covered by generated tests
+
+## Integration Testing Requirements
+
+**Integration Test Structure:**
+```go
+//go:build !account_level_tests  // or account_level_tests for account operations
```
Comment (sfc-gh-jmichalak) [inline .cursor/rules/sdk-development.mdc]: I think we can be more specific here: updating params, altering bcrs, etc

File: .cursor/rules/sdk-generator.mdc [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+    "Object",           // Struct name (singular)
+    g.KindOfT[SchemaObjectIdentifier](), // Identifier type
+).
+CreateOperation("https://docs.snowflake.com/...",
+    g.NewQueryStruct("CreateObject").
+        Create().
+        SQL("OBJECT").
+        Name().
+        OptionalComment(),
+).
```
Comment (sfc-gh-jmichalak) [inline .cursor/rules/sdk-generator.mdc]: What about alter and drop?

File: .cursor/rules/sdk-generator.mdc [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+  4. **Ask for help** if any links are broken or sections are missing
+  5. **Never proceed** with made-up fields or syntax
+- **Accuracy Requirements**: Only include fields and options explicitly documented in "Syntax" and "Output" sections
+- **Conservative Approach**: When in doubt, ask for clarification rather than making assumptions
+
+**DSL Patterns and DDL Tag System**
+
+The SDK uses a sophisticated DDL tag system to generate SQL from Go structs. Understanding these patterns is crucial for accurate implementation:
+
+**Core DDL Tags:**
```
Comment (sfc-gh-jmichalak) [inline .cursor/rules/sdk-generator.mdc]: I don't think we need this (and 3 sections below) here. IMO, we should only map the DSL to SQL, as the ddl tags are just an intermediary representation.

File: .cursor/rules/sdk-generator.mdc [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+
+**DSL Method Mapping:**
+- `SQL("TEXT")` → `ddl:"static" sql:"TEXT"`
+- `OptionalSQL("TEXT")` → `ddl:"keyword" sql:"TEXT"` (optional field)
+- `TextAssignment("FIELD", options)` → `ddl:"parameter" sql:"FIELD"`
+- `Name()` → `ddl:"identifier"` with appropriate identifier type
+- `OptionalComment()` → `ddl:"parameter,single_quotes" sql:"COMMENT"`
+
+**Complex Pattern Examples:**
+```go
```
Comment (sfc-gh-jmichalak) [inline .cursor/rules/sdk-generator.mdc]: I'd make this more descriptive, e.g: to generate a slice of strings, do...

File: .cursor/rules/sdk-generator.mdc [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+- **`DescriptionMappingKindSingleValue`**: Returns a single object (`*ObjectDetails`)
+  - **Use when**: DESCRIBE returns object metadata as a single record (name, owner, comment, etc.)
+  - **Generated method**: `Describe(ctx context.Context, id Identifier) (*ObjectDetails, error)`
+  - **Implementation**: Uses `validateAndQueryOne()` and `result.convert()`
+  - **Examples**: Tasks, Streams, Services, Git Repositories, Secrets
+
+- **`DescriptionMappingKindSlice`**: Returns a slice of objects (`[]ObjectProperty`)
+  - **Use when**: DESCRIBE returns property-value pairs or multiple configuration entries
+  - **Generated method**: `Describe(ctx context.Context, id Identifier) ([]ObjectProperty, error)`
+  - **Implementation**: Uses `validateAndQuery()` and `convertRows()`
```
Comment (sfc-gh-jmichalak) [inline .cursor/rules/sdk-generator.mdc]: Why do we need to list the implementation here? I'm quite confused about the purpose of this.

File: .cursor/rules/sdk-generator.mdc [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+
+### 1. Setup
+If missing, create definition files and add it to the mapping in `pkg/sdk/poc/main.go`:
+```go
+"your_object_def.go": sdk.YourObjectDef,
+```
+
+### 2. Documentation Analysis
+**Critical Phase - Must Be Thorough:**
+- Request documentation URLs for ALL operations (CREATE, ALTER, DROP, SHOW, DESCRIBE)
```
Comment (sfc-gh-jmichalak) [inline .cursor/rules/sdk-generator.mdc:254]: Q: have you tested passing only one doc? Can it traverse other pages on its own?

File: .cursor/rules/sdk-generator.mdc [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+### 3. Implementation and Verification
+**Based on Documentation Analysis:**
+- Map each documented syntax element to appropriate DSL pattern
+- Reference similar existing objects for complex patterns
+- When uncertain about DSL mapping, ask for guidance rather than guessing
+
+### 4. Generation and Validation
+```bash
+make run-generator-<object_name>  # Uses filename without _def suffix
+```
```
Comment (sfc-gh-jmichalak) [inline .cursor/rules/sdk-generator.mdc:274]: Let's also add that cursor should revert the changes overriding the `// Adjusted manually` comments.

**Review by sfc-gh-jmichalak — CHANGES_REQUESTED** [review]

File: .cursor/rules/sdk-development.mdc (hunk 1 above)
Comment (sfc-gh-jcieslak) [reply .cursor/rules/sdk-development.mdc]: On line 144, I mentioned to use `!account_level_tests` by default. For now, I would adjust it and remove this comment, as there are more layers in choosing the correct build tag. Starting with the most common one is a good start, I think.

File: .cursor/rules/sdk-generator.mdc (hunk 2 above)
Comment (sfc-gh-jcieslak) [reply .cursor/rules/sdk-generator.mdc]: Good point added

File: .cursor/rules/sdk-generator.mdc (hunk 5 above)
Comment (sfc-gh-jcieslak) [reply .cursor/rules/sdk-generator.mdc]: You're right they can be removed as they are auto-generated

File: .cursor/rules/sdk-generator.mdc (hunk 6 above)
Comment (sfc-gh-jcieslak) [reply .cursor/rules/sdk-generator.mdc:254]: Yes, I was testing it with a few different object types, thus different documentation pages. I was testing it mostly, I think, with https://docs.snowflake.com/en/sql-reference/commands-dbt-projects-on-snowflake, which even has one custom operation, and it was handling it pretty well.

File: .cursor/rules/sdk-generator.mdc (hunk 3 above)
Comment (sfc-gh-jcieslak) [reply .cursor/rules/sdk-generator.mdc]: Adjusted, Tag details should be extracted from sql_builder.go

File: .cursor/rules/sdk-generator.mdc (hunk 4 above)
Comment (sfc-gh-jcieslak) [reply .cursor/rules/sdk-generator.mdc]: General common patterns should be based on the current definitions, so adjusted (the listed examples were not that specific anyway).

File: .cursor/rules/sdk-generator.mdc (hunk 7 above)
Comment (sfc-gh-jcieslak) [reply .cursor/rules/sdk-generator.mdc:274]: Good point, added pre and post generation checks

File: .cursor/rules/sdk-development.mdc (hunk 1 above)
Comment (sfc-gh-jmichalak) [reply .cursor/rules/sdk-development.mdc]: To clarify: we can specify what are the exact conditions of having the account_level_tests tag. I mean that account_level_tests must be marked if we either read/update account params, alter bcrs, etc.

We can do this later, of course.

**Review by sfc-gh-jmichalak — APPROVED** [review]
